File: swift/p2044.swift
class Solution {
    func countMaxOrSubsets(_ nums: [Int]) -> Int {
        var maxOr = 0

        // Calculate the maximum OR value possible
        for num in nums {
            maxOr |= num
        }

        // Helper function to perform backtracking
        func backtrack(_ i: Int, _ currOr: Int) -> Int {
            if i == nums.count {
                return currOr == maxOr ? 1 : 0
            }

            
            return (
                // Include nums[i] in the subset
                backtrack(i + 1, currOr | nums[i]) +
                
                // Exclude nums[i] from the subset
                backtrack(i + 1, currOr))
        }

        // Start backtracking from the 0th element
        return backtrack(0, 0)
    }
}

File: python/p2044.py
from typing import List

class Solution:
    def countMaxOrSubsets(self, nums: List[int]) -> int:
        max_or = 0
        
        for num in nums:
            max_or |= num
        
        def backtrack(i, curr_or):
            if i == len(nums):
                return 1 if curr_or == max_or else 0
            
            return (
                # Include nums[i] in the subset
                backtrack(i + 1, curr_or | nums[i]) +
                # Exclude nums[i] from the subset
                backtrack(i + 1, curr_or)
            )
        
        return backtrack(0, 0)